pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -1052,10 +1075,26 @@ func (a *ApplicationBackupController) backupVolumes(backup *stork_api.Applicatio
 					logrus.Debugf("skipping driver %v for status check", driverName)
 					continue
 				}
-				status, err := driver.GetBackupStatus(backup)
-				if err != nil {
-					log.ApplicationBackupLog(backup).Errorf("failed to get vol status fro driver %v: %v", driverName, err)
-					return err
+				batchCount := defaultGetBackupStatusBatchCount
+				volumes := backup.Status.Volumes
+				status := make([]*stork_api.ApplicationBackupVolumeInfo, 0)
+				// Fetch the status of the volumes in batches and update the backup CR with last update timestamp
+				// this avoids backup timeout issue from px-backup
+				for i := 0; i < len(volumes); i += batchCount {
+					batchVolumes := volumes[i:min(i+batchCount, len(volumes))]
+					backup.Status.Volumes = batchVolumes
```

Comment:
move this code block also to the separated func:
```
batchVolumeInfos, err := a.processVolumeBatch(backup, driver, batchVolumes)
```

Reply:
Fixed , moved to a function